### ISSUE TITLE
FontAwesome V5

### ISSUE THREAD

**Issue by Miniland1333** [LibreTexts/ckeditor-binder-plugin#157]
Currently, this plugin is using the older FontAwesome v4 instead of the newer v5. The newer version has a significant rewrite under the hood and contains new icons that Delmar wants to use for other things.

FontAwesome v4 is currently being loaded within ckeditor-blinder-plugin at: https://github.com/LibreTexts/ckeditor-binder-plugin/blob/staging/src/scripts/registerPlugin.js#L13

Since FontAwesome is also shared by a few other pieces of code/HTML, do you think that it would be a good idea to make FontAwesome an external dependency rather than the current method of loading it within this plugin?

**Comment by sandertyu** [comment]
I believe we could do that for libretexts.org. The fontawesome script is necessary for many jupyterlab widgets to load properly, so if we remove it from the plugin then the plugin will not work the same when standing alone. Then again, we can probably just manually include the font awesome script on any html page we test it with that isn't libretexts.org.

Are you suggesting we remove the dependency from the plugin entirely, and just use the one provided by libretexts.org? We can probably at least try that.

**Comment by Miniland1333** [comment]
Yes, that does sound like what I would like to try for staging. One thing for us to remember is that we will likely need to use the v4 shims to make sure the plugins don't experience any compatibility issues. Let me know if you need help testing this on staging/Query and this is not urgent.
https://cdnjs.com/libraries/font-awesome

**Comment by moorepants** [comment]
The plugin should work outside of libretexts.org too. So, however you set this up, it would need to skip loading things if on that particular website.

**Comment by sandertyu** [comment]
I have removed font-awesome from `src/scripts/registerPlugin.js` and included some additional scripts to the `src/index.html` for independent testing. This is on staging and has been deployed to query. Removing font-awesome seems to have mostly not effected anything, but [this widget](https://query.libretexts.org/Sandboxes/jupyterteam_at_ucdavis.edu) is not displaying the proper button symbols on both libretexts and in our development page;

![matplotlib](https://cdn.discordapp.com/attachments/763198235237154837/858615016965144576/unknown.png)

**Comment by moorepants** [comment]
I don't think this is a good idea, in general.

**Comment by Miniland1333** [comment]
https://query.libretexts.org/Sandboxes/hdagnew@ucdavis.edu/Jupyter_Widgets

![image](https://user-images.githubusercontent.com/12994749/124042658-2dc79400-d9be-11eb-81aa-bbf89c031438.png)

Okay so I was able to get FontAwesome v5 working with the shim.

**Comment by moorepants** [comment]
I'm no longer seeing the icons on the buttons on query:
![image](https://user-images.githubusercontent.com/276007/127154739-07727048-6665-45a5-9ef6-acf97f685cc4.png)


https://query.libretexts.org/Sandboxes/jupyterteam_at_ucdavis.edu

**Comment by moorepants** [comment]
![image](https://user-images.githubusercontent.com/276007/127154878-2ddaef33-8bfc-480b-b832-5c0a69fc4745.png)

**Comment by moorepants** [comment]
The matplotlib example on query.libretexts.org now works and has the icons on the buttons.

**Comment by sandertyu** [comment]
We have merged in the changes suggested by Henry to load fontawesome based on an if statement which checks for certain tags within the HTML of the webpage. However, fontawesome does not load on its own when just using the local development version. We'll want to figure out why this is happening, and how to bundle fontawesome with this plugin regardless of context.

**Comment by moorepants** [comment]
@Miniland1333 it doesn't seem your fixed worked for the versions not loaded on libretexts. Can you have a look?

**Comment by Miniland1333** [comment]
If you are referring to how the changes aren't showing up when using `yarn serve`, my guess is that the different webpack configs have different entry-points. When using prod, it uses `registerPlugin.js` which includes these changes, but during development it uses `index.js` which does not.

This is evidenced by the following config that is only in webpack-production:
```javascript
  entry: {
    registerPlugin: Path.resolve(__dirname, '../src/scripts/registerPlugin.js'),
  },
```

**Comment by moorepants** [comment]
Thanks Henry. With the change you made, font awesome should be loaded by the plugin in production and development environments (if not on libretexts).

**Comment by Miniland1333** [comment]
I don't have time to test it, but you should be able to copy over the lines involving the loadScript() from registerPlugin.js to index.js so that it activates in both production and development.

**Comment by TimStewartJ** [comment]
I'll go ahead and close this, we can use #171 to use FA5 in local environments. I think its a better idea to keep the loading in registerPlugin.

**Comment by Miniland1333** [comment]
Closed by #171
